sql/core/src/main/scala/org/apache/spark/sql/parquet/ParquetTableSupport.scala

```diff
@@ -369,9 +371,6 @@ private[parquet] class MutableRowWriteSupport extends RowWriteSupport {
       case DateType => writer.addInteger(record.getInt(index))
       case TimestampType => writeTimestamp(record.getLong(index))
       case d: DecimalType =>
-        if (d.precisionInfo == None || d.precisionInfo.get.precision > 18) {
-          sys.error(s"Unsupported datatype $d, cannot write to consumer")
-        }
```

Comment:
I don't think Parquet allow mixed decimal precision. However, Hive uses a default precision = 10 and a default scale = 0 when precision/scale information is missing. I also did the same thing in #6617.